src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Subject, Subscription } from 'rxjs';
+import { Course, isCommunicationEnabled } from 'app/entities/course.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { ButtonSize } from 'app/shared/components/button.component';
+import { EventManager } from 'app/core/util/event-manager.service';
+import {
+    faChartBar,
+    faClipboard,
+    faComments,
+    faEye,
+    faFilePdf,
+    faFlag,
+    faGraduationCap,
+    faHeartBroken,
+    faListAlt,
+    faPersonChalkboard,
+    faTable,
+    faTimes,
+    faUserCheck,
+    faWrench,
+} from '@fortawesome/free-solid-svg-icons';
+import { FeatureToggle } from 'app/shared/feature-toggle/feature-toggle.service';
+import { CourseAdminService } from 'app/course/manage/course-admin.service';
+
+@Component({
+    selector: 'jhi-course-management-tab-bar',
+    templateUrl: './course-management-tab-bar.component.html',
+    styleUrls: ['./course-management-tab-bar.component.scss'],
+})
+export class CourseManagementTabBarComponent implements OnInit, OnDestroy {
+    readonly FeatureToggle = FeatureToggle;
+    readonly ButtonSize = ButtonSize;
+
+    course?: Course;
+
+    private paramSub: Subscription;
+    private courseSub: Subscription;
+
+    private dialogErrorSource = new Subject<string>();
+    dialogError$ = this.dialogErrorSource.asObservable();
+
+    // Icons
+    faTimes = faTimes;
+    faEye = faEye;
+    faWrench = faWrench;
+    faTable = faTable;
+    faUserCheck = faUserCheck;
+    faFlag = faFlag;
+    faListAlt = faListAlt;
+    faChartBar = faChartBar;
+    faFilePdf = faFilePdf;
+    faComments = faComments;
+    faClipboard = faClipboard;
+    faGraduationCap = faGraduationCap;
+    faHeartBroken = faHeartBroken;
+    faPersonChalkboard = faPersonChalkboard;
+
+    readonly isCommunicationEnabled = isCommunicationEnabled;
+
+    constructor(
+        private eventManager: EventManager,
+        private courseManagementService: CourseManagementService,
+        private courseAdminService: CourseAdminService,
+        private route: ActivatedRoute,
+        private router: Router,
+    ) {}
+
+    /**
+     * On init load the course information and subscribe to listen for changes in courses.
+     */
+    ngOnInit() {
+        let courseId = 0;
+        this.paramSub = this.route.params.subscribe((params) => {
+            courseId = params['courseId'];
+        });
+
+        this.courseSub = this.courseManagementService.find(courseId).subscribe((courseResponse) => {
```

Comment:
Shouldn't this code be placed inside the courseId-Subsciption? We first need to get the id and only execute this afterwards, right?

Reply:
Fixed